plugins/inputs/nstat/nstat.go

```diff
@@ -83,14 +83,16 @@ func (ns *Nstat) Gather(acc telegraf.Accumulator) error {
 		return err
 	}
 
-	// collect SNMP6 data
-	snmp6, err := ioutil.ReadFile(ns.ProcNetSNMP6)
-	if err != nil {
-		return err
-	}
-	err = ns.gatherSNMP6(snmp6, acc)
-	if err != nil {
-		return err
+	// collect SNMP6 data, if SNMP6 directory exists (IPv6 enabled)
+	if _, err := os.Stat(ns.ProcNetSNMP6); err == nil {
+		snmp6, err := ioutil.ReadFile(ns.ProcNetSNMP6)
```

Comment:
This introduces a race condition between the `os.Stat` call and the `os.Open` call within `ioutil.ReadFile`.  This is one of those situations where you need to ask forgiveness instead of permission, can you try this:
```go
	snmp6, err := ioutil.ReadFile(ns.ProcNetSNMP6)
	if err != nil && !os.IsNotExist(err) {
		return err
	}
```

Reply:
Right, good catch. I've refactored the code.